app/lib/services/funding_eligibility.rb

```diff
@@ -35,9 +48,11 @@ def funded?
 
     def funding_eligiblity_status_code
       @funding_eligiblity_status_code ||= begin
-        return NO_INSTITUTION if institution.nil?
+        return NO_INSTITUTION if institution.nil? && !lead_mentor_course? && !approved_itt_provider
         return PREVIOUSLY_FUNDED if previously_funded?
-        return FUNDED_ELIGIBILITY_RESULT if eligible_urns.include?(institution.urn)
+        return FUNDED_ELIGIBILITY_RESULT if eligible_urns.include?(institution.try(:urn))
+        return FUNDED_ELIGIBILITY_RESULT if lead_mentor_course? && approved_itt_provider && lead_mentor
+        return NOT_LEAD_MENTOR_COURSE if approved_itt_provider && lead_mentor
```

Comment:
This section has become confusing to me as the guard clauses are relying on each others results and the lead mentor checks are weaving into non-lead mentor results. It works, but it's harder to parse than it has to be.

It might be clearer to rework all the above into something like this so that if you have chosen a lead mentor, you get diverted to a whole set of separate status code calculations that can be reasoned about. It also means that the rest of the guard clauses remain untouched.

```
def funding_eligiblity_status_code
  @funding_eligiblity_status_code ||= begin
    return lead_mentor_eligibility_status_code if lead_provider_status_code_calculations_active?

    return NO_INSTITUTION if institution.nil?
    return PREVIOUSLY_FUNDED if previously_funded?
    return FUNDED_ELIGIBILITY_RESULT if eligible_urns.include?(institution.try(:urn))

    ...
  end
end

def lead_provider_status_code_calculations_active?
  # I'm not sure the exact trigger for this, I think it would be:
  lead_mentor
end

def lead_mentor_eligibility_status_code
  return NOT_LEAD_MENTOR_COURSE unless lead_mentor_course? # Don't do a check to the API unless necessary
  return PREVIOUSLY_FUNDED if previously_funded?
  return FUNDED_ELIGIBILITY_RESULT if approved_itt_provider

  # This is a new status code, currently the code reverts to INELIGIBLE_INSTITUTION_TYPE which is less specific and clear what happened
  return NOT_APPROVED_IIT_PROVIDER_RESULT
end
```

The above rules and checks may not be exactly what they need to be but I think it's important to not weave the two sets of checks together as it feels like the beginning of the road of making this section hard to keep straight in your head.

Reply:
I agree and wanted to suggest this code needs more attention, do you feel this is a blocker for this release?